source/Patches/Shuffle.cs

```diff
@@ -7,21 +7,19 @@ public static class ListExtensions
     {
         public static void Shuffle<T>(this List<T> list)
         {
-            var count = list.Count;
-            var last = count - 1;
-            for (var i = 0; i < last; ++i)
+            for (var i = list.Count-1; i > 0; --i)
             {
-                var r = Random.Range(i, count);
-                var tmp = list[i];
-                list[i] = list[r];
-                list[r] = tmp;
+                var j = Random.Range(0, i+1);
+                (list[i], list[j]) = (list[j], list[i]);
             }
         }
 
         public static T TakeFirst<T>(this List<T> list)
         {
+            list.Shuffle();
```

Comment:
Don't think these are needed anymore.

Reply:
it work better now

Reply:
I was testing without them, and it worked fine. I don't think it's a good idea to introduce this behavior to `TakeFirst`, which is a generic method used for any List, not just the ones in `GenEachRole`.

Reply:
point taken, will remove them then